pkg/cluster/template/scripts/pd.go

```diff
@@ -142,58 +145,7 @@ func NewPDScaleScript(name, ip, deployDir, dataDir, logDir string) *PDScaleScrip
 	return &PDScaleScript{*NewPDScript(name, ip, deployDir, dataDir, logDir)}
 }
 
-// WithScheme set Scheme field of PDScaleScript
-func (c *PDScaleScript) WithScheme(scheme string) *PDScaleScript {
-	c.Scheme = scheme
-	return c
-}
-
-// WithClientPort set ClientPort field of PDScaleScript
-func (c *PDScaleScript) WithClientPort(port int) *PDScaleScript {
-	c.ClientPort = port
-	return c
-}
-
-// WithPeerPort set PeerPort field of PDScript
-func (c *PDScaleScript) WithPeerPort(port int) *PDScaleScript {
-	c.PeerPort = port
-	return c
-}
-
-// WithNumaNode set NumaNode field of PDScaleScript
-func (c *PDScaleScript) WithNumaNode(numa string) *PDScaleScript {
-	c.NumaNode = numa
-	return c
-}
-
-// AppendEndpoints add new PDScaleScript to Endpoints field
-func (c *PDScaleScript) AppendEndpoints(ends ...*PDScript) *PDScaleScript {
-	c.Endpoints = append(c.Endpoints, ends...)
-	return c
-}
-
-// WithListenHost set listenHost field of PDScript
-func (c *PDScaleScript) WithListenHost(listenHost string) *PDScaleScript {
-	c.ListenHost = listenHost
-	return c
-}
-
 // Config generate the config file data.
 func (c *PDScaleScript) Config() ([]byte, error) {
-	fp := path.Join("templates", "scripts", "run_pd_scale.sh.tpl")
-	log.Infof("script path: %s", fp)
-	tpl, err := embed.ReadFile(fp)
-	if err != nil {
-		return nil, err
-	}
-	return c.ConfigWithTemplate(string(tpl))
-}
-
-// ConfigToFile write config content to specific path
-func (c *PDScaleScript) ConfigToFile(file string) error {
-	config, err := c.Config()
-	if err != nil {
-		return err
-	}
-	return os.WriteFile(file, config, 0755)
+	return c.configWithScript("run_pd_scale.sh.tpl")
```

Comment:
I think this may not work as you expected since you removed `func (c *PDScaleScript) WithXXX`.

When we call WithXXX [in scale out stage](https://github.com/pingcap/tiup/blob/master/pkg/cluster/spec/pd.go#L269), the `WithXXX` will return a PDScript pointer because you removed `WithXXX` on `PDScaleScript` (it will fallback to PDScript's WithXXX).  Then when you call [ConfigToFile](https://github.com/pingcap/tiup/blob/master/pkg/cluster/spec/pd.go#L280) or `Config`, it will call on `PDScript`, so the script generated in scale-out stage is not correct.

Reply:
Thanks for your professional advice 👍

Reply:
PTAL

Reply:
Still not work because PDScaleScript not implements [ConfigToFile](https://github.com/pingcap/tiup/pull/1253/files#diff-52ed75eb339663407c56ff6cf1b6bdf3a0995e7e9da391e4d28f113c9f14eb39R281)

Reply:
Sorry, please take another look.